datafu-spark/src/main/scala/spark/utils/overwrites/SparkPythonRunner.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy
+
+import java.io._
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+import datafu.spark.ScalaPythonBridge
+import org.apache.log4j.Logger
+
+import org.apache.spark.api.python.PythonUtils
+import org.apache.spark.util.Utils
+
+/**
+ * We wrap Spark's PythonRunner because we failed on premature python process closing.
```

Comment:
This documentation is confusing.  It doesn't really explain what this does (more like explains background on why it was written, which is still useful but should be at bottom of doc).  Also not clear how this is wrapping PythonRunner because only PythonRunner.formatPath is called.  Is this a copy of the code from PythonRunner that was modified?

Reply:
yes. this is a copy from PythonRunner that was modified.
i'll explain that.
also it is in folder called "spark/utils/overwrites" for clarity.